source/base/index_set.cc

```diff
@@ -63,7 +63,11 @@ IndexSet::IndexSet (const Epetra_Map &map)
 IndexSet::IndexSet (const Epetra_Map &map)
   :
   is_compressed (true),
-  index_space_size (map.NumGlobalElements()),
+#ifndef DEAL_II_WITH_64BIT_INDICES
+  index_space_size (1+map.MaxAllGID()),
+#else
+  index_space_size (1+map.MaxAllGID64()),
+#endif
```

Comment:
want to add an assert to the body of the constructor that the MinAllGID is zero?